mitodl/edx-platform
# Re-implemented remote gradebook enrollment merge/overload

#### What are the relevant tickets?

Closes #14
⚠️  REBASED ON #15 ⚠️ 

#### What's this PR do?

- Reimplements remote gradebook merge/overload functionality (shown in this issue: https://github.com/mitocw/edx-platform/issues/76; live example: https://staging.mitx.mit.edu/courses/DevOps/0.001/2015_Summer/legacy_instructor_dash)

#### Where should the reviewer start?

Probably `lms/djangoapps/instructor/views/api.py` and `lms/static/js/instructor_dashboard/remote_gradebook.js`

#### How should this be manually tested?

- Set the `ENABLE_INSTRUCTOR_REMOTE_GRADEBOOK_CONTROLS ` feature to `True`
- Edit your settings (ask me or Peter for the proper settings here):
  - `REMOTE_GRADEBOOK` setting in `lms.env.json` should have a `URL` and `DEFAULT_NAME` property
  - `lms.auth.json` should have `REMOTE_GRADEBOOK_USER` and `REMOTE_GRADEBOOK_PASSWORD` settings
- Navigate to the `Remote Gradebook` tab for some course (eg: `http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-remote_gradebook` - make sure you have staff/admin privileges)
- List the users in some section to see who's in there
- Click 'Merge enrollment list' and inspect the results. Make sure that the results seem to line up with expectations
- Click 'Overload enrollment list', confirm that you get a warning about un-enrolling existing users, and click OK. Make sure that the results seem to line up with expectations
